coco docs: clarify platforms page and agent policy (#479)
NVIDIA/cloud-native-docs: confidential-containers/configure-workloads.rst

A Confidential Container workload is a standard Kubernetes pod that runs inside a TEE-protected
virtual machine and requests one or more GPUs through the NVIDIA Kata sandbox device plugin.
Compared with a traditional GPU pod, a Confidential Container workload pod manifest differs in
the following ways:

* It selects a TEE-aware Kata runtime class instead of the default ``runc``-based runtime.
* It requests GPU and NVSwitch resources using the resource types advertised by the NVIDIA
  Kata sandbox device plugin, which can be either default names or model-specific names.
* For NVSwitch-based HGX systems, it requests every GPU and NVSwitch on the node together so
  that all devices reside inside the same Confidential Container virtual machine.
* For an attested production deployment, it includes a Kata agent security policy that limits

>   which Agent API calls the untrusted host can make into the guest.

Author: This is a little outside the bounds of the PR, but continuing my penchant for hunting pronouns for sport, could "it" in these bullets be replaced with "the manifest"?

As a newcomer, this list is teeming with terms and I had to backtrack to remind myself what "it" was.

Reviewer: agreed - I am addressing this as well for the three other bullet points listed above

      $ kubectl delete -f multi-gpu-kata.yaml

> .. _coco-agent-security-policy:

Reviewer 2: nit: any reason this tag is coco-agent instead of kata-agent?
not a big deal since this string doesn't appear to the user, but it could confound refactoring in the future

***********************************
Attach a Kata Agent Security Policy
***********************************

The Kata agent runs inside the guest virtual machine and manages the container lifecycle.
Because the Kata shim on the host is outside the TEE, the host can still issue Agent API calls

> into the guest unless you restrict those Agent API calls.

Author: Q: does an "untrusted host" belong somewhere in here?  I thought I read that before and that's what gets this human's attention.  Is this correct?

```suggestion
Because the Kata shim on the host is outside the TEE, the untrusted host can still issue Agent API calls
into the guest unless you restrict those Agent API calls.
```

calls, encodes the policy in base64, and appends it as an annotation on the same file.

#. Download ``genpolicy`` from the latest Kata Containers release that is compatible with this
   reference architecture.
   For usage details, refer to the
   `Agent Policy generation tool <https://github.com/kata-containers/kata-containers/blob/main/src/tools/genpolicy/README.md>`_
   documentation.

#. Run ``genpolicy`` against the manifest to deploy:

   .. code-block:: console

      $ genpolicy -y cuda-vectoradd-kata.yaml

   Review the generated policy before you apply the manifest.
   The automatically-generated policy is a starting point.
   The policy can allow operations you do not want, such as ``kubectl exec``, or omit operations your
   workload needs.

   .. important::

      Review and, if needed, edit the policy so that it matches your threat model before you
      use it.

#. Apply the annotated manifest:

   .. code-block:: console

      $ kubectl apply -f cuda-vectoradd-kata.yaml

**********
Next Steps
**********

* Refer to :doc:`Attestation <attestation>` for Trustee concepts and a local connectivity test.

>   A complete attestation process should cover the Kata Agent API surface as well as the TEE.

Reviewer 2: ```suggestion
  A complete attestation policy should cover the Kata Agent API surface as well as the TEE.
```

  The agent security policy contained in the workload annotation is included in the measured
  payload that is attested with the guest.
  The agent security policy is distinct from the Trustee attestation policy that decides whether

>   to release secrets.

Reviewer 2: I don't know an easier way to phrase this but there are 3 policies. Their interrelation is captured upstream here: 
https://confidentialcontainers.org/docs/getting-started/securing-workloads/#understanding-coco-policies

      $ kubectl delete -f multi-gpu-kata.yaml

.. _coco-agent-security-policy:

***********************************
Attach a Kata Agent Security Policy
***********************************

The Kata agent runs inside the guest virtual machine and manages the container lifecycle.
Because the Kata shim on the host is outside the TEE, the host can still issue Agent API calls
into the guest unless you restrict them.

> An agent security policy is a Rego policy that the agent enforces so that only the operations

Author: sugg: newly introduced or defined terms are italicized on first use--that's the "hey, this bit is important" signal.

```suggestion
An _agent security policy_ is a Rego policy that the agent enforces so that only the operations
```

your workload declared are allowed.
That restriction is required for an attested, secure end-to-end deployment: attestation proves
the TEE before secrets are released, and the agent policy limits what the untrusted host can
ask the guest to do afterward.

This policy is not the Trustee attestation policy that decides whether to release secrets.
An attested deployment uses both.
Refer to :doc:`Attestation <attestation>` for Trustee concepts and a local connectivity test.

The :doc:`sample workload <run-sample-workload>` omits an agent policy so that you can confirm
GPU passthrough with the smallest possible manifest.
Do not treat that YAML as a production template.

The Kata Containers ``genpolicy`` tool reads your Kubernetes YAML, infers the intended Agent API
calls, encodes the policy in base64, and appends it as an annotation on the same file.

> #. Obtain ``genpolicy`` from the latest Kata Containers release that is compatible with this

Author: Maybe?  Obtain feels a little wordy.  Not critical.

```suggestion
#. Download ``genpolicy`` from the latest Kata Containers release that is compatible with this
```

Reviewer: @mikemckiernan - yes. @fidencio, to me, download seems right here as genpolicy is published as part of kata releases - do you want to get a bit more specific here?

   reference architecture.
   For build and usage details, refer to the
   `Agent Policy generation tool <https://github.com/kata-containers/kata-containers/blob/main/src/tools/genpolicy/README.md>`_
   documentation.

> #. Run ``genpolicy`` against the manifest you will deploy, for example:

Author: nit: Use present tense whenever possible.  It's usually clearest.  In this case the loss of the second-person "you" is OK because the instructional text is imperative anyway.

```suggestion
#. Run ``genpolicy`` against the manifest to deploy:
```

   .. code-block:: console

      $ genpolicy -y cuda-vectoradd-kata.yaml

> #. Review the generated policy before you apply the manifest.

Author: nit: This is good instructional imperative text, but it seems (to me) to be part of the preceding step of generating the agent security policy, not a new task.

```suggestion
   Review the generated policy before you apply the manifest.
```

>    Auto-generated policy is a starting point.

Author: ```suggestion
   The automatically-generated policy is a starting point.
```

>    It can allow operations you do not want, such as ``kubectl exec``, or omit operations your

Author: nit: restating the noun is clearer than a pronoun

```suggestion
   The policy can allow operations you do not want, such as ``kubectl exec``, or omit operations your
```
